GitCommands/PathUtil.cs

```diff
@@ -159,7 +160,93 @@ private static string ResolveRelativePath(string path, string relativePath)
 
         internal static bool IsWslPath(string path)
         {
-            return path.ToLower().StartsWith(@"\\wsl$\");
+            return path.ToLower().StartsWith(WslPrefix);
+        }
+
+        /// <summary>
+        /// Get the name of the distribution (like "Ubuntu-20.04") for WSL2 paths.
+        /// </summary>
+        /// <param name="path">Directory path.</param>
+        /// <returns>Name of the distro or empty for non WSL2 paths</returns>
+        public static string GetWslDistro(string path)
+        {
+            int distroLen = GetWslDistroLength(path);
+            return distroLen <= 0 ? "" : path.Substring(WslPrefix.Length, distroLen);
+
+            static int GetWslDistroLength(string path)
+            {
+                if (string.IsNullOrWhiteSpace(path) || !IsWslPath(path))
+                {
+                    return -1;
+                }
+
+                return path.IndexOfAny(new[] { '\\', '/' }, WslPrefix.Length) - WslPrefix.Length;
+            }
+        }
+
+        /// <summary>
+        /// Convert a path for the Git executable.
+        /// For native (Windows) the path is the same, for WSL Git the path may be transformed.
+        /// </summary>
+        /// <param name="path">Path as seen by the Git Extensions native (Windows) executable.</param>
+        /// <param name="wslDistro">Name of the distro or empty for non WSL paths</param>
+        /// <returns>Posix path if not a WSL distro (Windows Git), WSL path for WSL Git.</returns>
+        public static string GetRepoPath(string? path, string wslDistro)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return path;
+            }
+
+            if (string.IsNullOrEmpty(wslDistro))
+            {
+                return path.ToPosixPath();
+            }
+
+            string pathDistro = GetWslDistro(path);
+            if (!string.IsNullOrEmpty(pathDistro))
+            {
+                // Unexpected but not handled as an error
+                if (pathDistro != wslDistro)
+                {
+                    return path.ToPosixPath();
+                }
+
+                return path[(WslPrefix.Length + pathDistro.Length)..].ToPosixPath();
+            }
+
+            if (path.Length > 2 && char.IsLetter(path[0]) && path[1] == ':')
+            {
+                return $"/mnt/{char.ToLower(path[0])}{path[2..].ToPosixPath()}";
+            }
+
+            return path.ToPosixPath();
+        }
+
+        /// <summary>
+        /// Convert a path to native (Windows) format.
+        /// </summary>
+        /// <param name="path">Path as seen by the Git executable, possibly WSL Git.</param>
+        /// <param name="wslDistro">Name of the distro or empty for non WSL paths.</param>
+        /// <returns>Path in in Windows format with native file separators.</returns>
+        public static string GetNativePath(string path, string wslDistro)
```

Comment:
Either incorrect nullable annotations, or absent null checks and redundant checks below.

Reply:
Changing to not return null

Reply:
We either need to annotate the args as `string?` or add `_ = path ?? throw new ANE(nameof(path))`

Reply:
The args are not null and should not be nullable
I will add a null check for path still